### PR TITLE
Reduce latency of cloning network sender using Arc pointers

### PR DESCRIPTION
**Pull request by vusirikala**
### Description
In order to reduce latency in the round manager, use Arc pointers for NetworkSender to reduce the latency of cloning the NetworkSender.

### Test Plan
Evaluate the latency of this change by running forge test with 100 nodes on this PR and compare the stats with the main branch.
The results can be compared here:
Forge test on this branch: https://github.com/aptos-labs/aptos-core/actions/runs/7965930582
Forge test on main branch: https://github.com/aptos-labs/aptos-core/actions/runs/7965951082

### CODE REVIEW

File: testsuite/forge-cli/src/main.rs [hunk 1]
```diff
@@ -976,7 +976,7 @@ fn realistic_env_sweep_wrap(
 }
 
 fn realistic_env_load_sweep_test() -> ForgeConfig {
-    realistic_env_sweep_wrap(20, 10, LoadVsPerfBenchmark {
+    realistic_env_sweep_wrap(100, 10, LoadVsPerfBenchmark {
```
Comment (sitalkedia) [inline testsuite/forge-cli/src/main.rs:979]: Please revert this before landing.

**Review by sitalkedia — APPROVED** [review]

File: consensus/src/round_manager.rs [hunk 2]
```diff
@@ -295,15 +295,14 @@ impl RoundManager {
...
-            let mut network = self.network.clone();
             #[cfg(feature = "failpoints")]
             {
                 if self.check_whether_to_inject_reconfiguration_error() {
                     self.attempt_to_inject_reconfiguration_error(&proposal_msg)
                         .await?;
                 }
             }
-            network.broadcast_proposal(proposal_msg).await;
+            self.network.clone().broadcast_proposal(proposal_msg).await;
```
Comment (bchocho) [inline consensus/src/round_manager.rs:305]: The clone seems unnecessary

File: consensus/src/round_manager.rs [hunk 3]
```diff
@@ -582,6 +581,7 @@ impl RoundManager {
 
         if self.sync_only() {
             self.network
+                .clone()
```
Comment (bchocho) [inline consensus/src/round_manager.rs:584]: same here

File: consensus/src/round_manager.rs [hunk 4]
```diff
@@ -622,7 +622,10 @@ impl RoundManager {
 
         self.round_state.record_vote(timeout_vote.clone());
         let timeout_vote_msg = VoteMsg::new(timeout_vote, self.block_store.sync_info());
-        self.network.broadcast_timeout_vote(timeout_vote_msg).await;
+        self.network
+            .clone()
```
Comment (bchocho) [inline consensus/src/round_manager.rs:626]: same here

File: consensus/src/round_manager.rs (hunk 2 above)
Comment (vusirikala) [reply consensus/src/round_manager.rs:305]: Done.

File: testsuite/forge-cli/src/main.rs (hunk 1 above)
Comment (vusirikala) [reply testsuite/forge-cli/src/main.rs]: Done.

File: consensus/src/round_manager.rs (hunk 3 above)
Comment (vusirikala) [reply consensus/src/round_manager.rs]: Done.

File: consensus/src/round_manager.rs (hunk 4 above)
Comment (vusirikala) [reply consensus/src/round_manager.rs:625]: Done.

**Pull request merged**